e2e/libs/volume/crd_volume.py

```diff
@@ -0,0 +1,299 @@
+        if volume_name == "" or size == "" or replica_count == "" or volume_type == "":
+            raise Exception(f"empty parameters: volume_name={volume_name}, size={size}, replica_count={replica_count}, volume_type={volume_type}")
+
+        logging.info(f"creating {size} Gi {volume_type} volume with {replica_count} replicas")
+
+        volume_body = {
+            "apiVersion": "longhorn.io/v1beta2",
+            "kind": "Volume",
+            "metadata": {"name": volume_name},
+            "spec": {
+                "frontend": "blockdev",
+                "replicaAutoBalance": "ignored",
+                "size": str(int(size) * Gi),
+                "numberOfReplicas": int(replica_count),
+                "accessMode": volume_type.lower()
+            }
+        }
+        logging.debug(f"volume creation body: {volume_body}")
+
+        api_response = self.obj_api.create_namespaced_custom_object(
+            group="longhorn.io",
+            version="v1beta2",
+            namespace="longhorn-system",
+            plural="volumes",
+            body=volume_body
+        )
+        logging.debug(f"response of volume creation {api_response}")
+
+    def attach(self, volume_name, node_name):
+        self.obj_api.patch_namespaced_custom_object(
+            group="longhorn.io",
+            version="v1beta2",
+            namespace="longhorn-system",
+            plural="volumes",
+            name=volume_name,
+            body={
+                    "spec": {
+                        "nodeID": node_name
+                    }
+                }
+        )
+
+        try:
+            self.obj_api.patch_namespaced_custom_object(
+                group="longhorn.io",
+                version="v1beta2",
+                namespace="longhorn-system",
+                plural="volumeattachments",
+                name=volume_name,
+                body={
+                        "spec": {
+                            "attachmentTickets": {
+                                "": {
+                                    "generation": 0,
+                                    "id": "",
+                                    "nodeID": node_name,
+                                    "parameters": {
+                                        "disableFrontend": "false",
+                                        "lastAttachedBy": ""
+                                    },
+                                    "type": "longhorn-api"
+                                }
+                            }
+                        }
+                    }
+            )
+        except ApiException as e:
+            if e.reason == "Not Found":
+                logging.warning(f'not support volumeattachments:', e)
+            else:
+                Exception(f'exception for creating volumeattachments:', e)
+
+        self.wait_for_volume_state(volume_name, "attached")
+
+    def get_node_without_replica(self, volume_name):
```

Comment:
[This function](https://github.com/longhorn/longhorn-tests/pull/1300/files#diff-aeab77e409abd8b958c29fbc6b54f9a1ff913cf0d3444ee3a1a58d6a00489e4dR137-R139) looks incomplete.

Reply:
The function `get_node_without_replica()` is replaced by `get_no_replica_node()`, so I will remove it. Thank you.